lib/route/origin_security_test.go

```diff
@@ -49,17 +37,57 @@ func TestOriginSecurity(t *testing.T) {
 			server := httptest.NewServer(adapter(nextHandler))
 			defer server.Close()
 
-			req, err := http.NewRequest("GET", server.URL, nil)
-			require.NoError(t, err)
-			req.Header.Add("Origin", tc.origin)
-			res, err := http.DefaultClient.Do(req)
-			require.NoError(t, err)
+			for _, goodOrigin := range tc.goodOrigins {
+				// Cross-site case
+				assertOriginAccepted(t, server, goodOrigin, goodOrigin)
+				assertOriginAccepted(t, server, goodOrigin, "")
+				// Same-origin case
+				assertOriginAccepted(t, server, "", goodOrigin)
+				// Undefined - referer and origin missing
+				assertOriginDenied(t, server, "", "")
```

Comment:
pull this one out of the loop?

Reply:
ah yes, the '(bad, bad)' one can come out too...